templates/rag-conversation-zep/pyproject.toml

```diff
@@ -0,0 +1,29 @@
+[tool.poetry]
+name = "rag_conversation_zep"
+version = "0.0.1"
+description = "A RAG application built with Zep. Zep provides a VectorStore implementation to the chain."
+authors = ["Daniel Chalef <daniel@getzep.com>"]
+readme = "README.md"
+
+[tool.poetry.dependencies]
+python = ">=3.8.1,<4.0"
+langchain = ">=0.0.313, <0.1"
+pydantic = ">1,<2"
```

Comment:
woah does this work when they're both exclusive? would've thought it needs to be >=1<2 or >1<=2

Reply:
When pydantic is unconstrained and 2.4.2 is installed, I get the following. Should langserve be compatible with 2.4?

```
Process SpawnProcess-1:
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/process.py", line 314, in _bootstrap
    self.run()
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/uvicorn/_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/uvicorn/server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 190, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.11/3.11.6/Frameworks/Python.framework/Versions/3.11/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/uvicorn/server.py", line 68, in serve
    config.load()
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/uvicorn/config.py", line 473, in load
    self.loaded_app = self.loaded_app()
                      ^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/langchain_cli/dev_scripts.py", line 44, in create_demo_server_configurable
    return create_demo_server(config_keys=["configurable"])
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/langchain_cli/dev_scripts.py", line 30, in create_demo_server
    add_routes(
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/langserve/server.py", line 407, in add_routes
    input_type_ = _resolve_model(runnable.get_input_schema(), "Input", model_namespace)
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 1125, in get_input_schema
    return self.first.get_input_schema(config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 2505, in get_input_schema
    return super().get_input_schema(config)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/langchain/schema/runnable/base.py", line 178, in get_input_schema
    return create_model(
           ^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/pydantic/v1/main.py", line 1026, in create_model
    return meta(__model_name, resolved_bases, namespace, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/pydantic/v1/main.py", line 197, in __new__
    fields[ann_name] = ModelField.infer(
                       ^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/pydantic/v1/fields.py", line 506, in infer
    return cls(
           ^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/pydantic/v1/fields.py", line 436, in __init__
    self.prepare()
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/pydantic/v1/fields.py", line 557, in prepare
    self.populate_validators()
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/pydantic/v1/fields.py", line 831, in populate_validators
    *(get_validators() if get_validators else list(find_validators(self.type_, self.model_config))),
                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/danielchalef/Library/Caches/pypoetry/virtualenvs/rag-conversation-zep-jrYlkh3b-py3.11/lib/python3.11/site-packages/pydantic/v1/validators.py", line 765, in find_validators
    raise RuntimeError(f'no validator found for {type_}, see `arbitrary_types_allowed` in Config')
RuntimeError: no validator found for <class 'rag_conversation_zep.chain.ChatHistory'>, see `arbitrary_types_allowed` in Config
```

Reply:
I can fix this quickly - pydantic compatibility is quite the challenge sometimes!

I'll swap out `from pydantic` for `from langchain.pydantic_v1`, which we are using in most templates for now.

I think Bagatur's comment was more about the `1<x<2` relation, suggesting `1<=x<2`.

Reply:
Great, thanks!